packages/rpc/json/error.ts

```diff
@@ -0,0 +1,27 @@
+export class JsonRpcProviderError extends Error {
+  public readonly code: number
+  public readonly data?: unknown
+
+  public static wrap(error: { message: string; code: number; data?: unknown }) {
+    return new JsonRpcProviderError(error.message, error.code, error.data)
+  }
+
+  public constructor(message: string, code: number, data?: unknown) {
+    super(message)
+    this.name = "JsonRpcProviderError"
+    this.code = code
+    this.data = data
+
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, JsonRpcProviderError)
+    }
+  }
+
+  public unwrap() {
+    return {
+      message: this.message,
+      code: this.code,
+      data: this.data
+    }
```

Comment:
The format should be 

```
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -32001,
        "message": "..."
    }
}
```

Reply:
[conform with rpc format](https://github.com/pilagod/waallet-extension/pull/179/commits/06d0d2f602ff37860667c21498090002cac71389)